### DISCUSSION TITLE
lax.js client-side plugin - "window.lax is undefined"

### DISCUSSION THREAD

**Discussion by thely** [nuxt/nuxt#9707] [Questions (Nuxt 2)]
I love the look of lax.js, and it offers things that some other parallax libraries don't. Problem is, I can't get it working with Nuxt! I keep getting `window.lax is undefined` no matter how I try to client-side-plugin this library.

### Versions

nuxt: 2.15.3

### Reproduction

Make a plugin to handle the library, add it to the Vue prototype, no dice.

`@/plugins/lax.js`
```js
import Vue from "vue";
import lax from "/node_modules/lax.js/lib/lax.js";

Vue.prototype.$lax = lax;
```

registering the plugin in `nuxt.config.js:`
```js
plugins: [
    { src: "@/plugins/lax.js", mode: "client" },
  ],
```

in a component:
```js
export default {
  mounted() {
    if (process.client) {
      window.onNuxtReady((e) => {
        this.$lax.init(); // fails here
      });
      this.$lax.addDriver("scrollY", function () {
        return window.scrollY;
      });
    }
  },
};
```

### Steps to reproduce
As above.

### What is Expected?
By doing as many things as I possibly can to ensure that the window exists when `lax.init()` is called, I would expect the `window` to exist by now. But apparently, it doesn't!

I've also tried this where I load it on the component itself instead of using a plugin (`require()` it on mounted()`), still saying `window.lax is undefined`. I have no idea if this is a fault of the lax.js library, but I figured I'd eliminate the Nuxt part of it first (since that's the part I have control over!).

### What is actually happening?
`window.lax is undefined`.

**Comment by danielroe** [comment]
@thely Likely the issue is the import into your client plugin. Although the plugin is client-only, it is still imported into your server build, just not executed. Try using a dynamic import within the function body instead. For example:
```js
import Vue from "vue";

export async default () => {
  const lax = await import("lax.js/lib/lax.js");

  Vue.prototype.$lax = lax;
}
```
